tools/test-proxy/documentation/integration-guidelines.md

```diff
@@ -0,0 +1,52 @@
+# Test-Proxy Integration Guidelines
+
+This document is intended to make help users integrate the test-proxy with their specific language. While it is not intended to get into extreme technical details, it should definitely go far enough to direct where necessary.
+
+## General Guidance
+
+Every language will need to follow the same basic process.
+
+1. Read the test-proxy readme.
+2. Spin up a local version of the test-proxy
+3. Transition a single test to flow through the test-proxy
+4. Submit a PR
+5. Add necessary additions for the test-proxy to start up in CI
```

Comment:
Could we add a link to languages that already have this finished? Go (and probably other languages) has a simple one line addition to the ci.yml